src/kernel/messages.c

```diff
@@ -194,13 +194,15 @@ message *msg_message(const char *name, const char *sig, ...)
         }
         else {
             log_error("invalid parameter %s for message type %s\n", paramname, mtype->name);
+            assert(!"invalid parameter for message type");
```

Comment:
So ein assert mag ich eigentlich nicht. In der Praxis bedeutet das, wenn ein Message-Parameter falsch ist, dann gibt es keine Auswertung. Die bessere Lösung ist da, dass man log_error macht wie hier, und einen Workaround für fehlende Daten macht (z.B. Default-Werte oder die Message unterdrücken). Ich kümmere mich um ERROR Einträge im Log in der Regel innerhalb einer Woche.
Zweiter Vorteil davon: Man kann den Code dann auch unit-testen (schauen, dass der Fehler im Log auftaucht, oder dass der Workaround tut, was wir erwarten).
So ein assert ist für harte Fälle, wo man ausdrücken will, dass entweder der Benutzer einer Funktion sie falsch verstanden hat, und falsche Argumente übergeben hat, oder wo es nicht klar ist, wie man einen Workaround machen sollte (wenn man z.B. einen Null-Pointer gekriegt hat, und das Einheitengewicht bestimmen soll). Und natürlich sollte ein assert niemals durch Befehle von Spielern ausgelöst werden können, indem man z.B. ein assert über die Länge eines Befehls-Keywortes macht oder so etwas.